src/views/Farms/components/Table/Table.tsx

```diff
@@ -0,0 +1,291 @@
+        padding-right: 0;
+        background: ${(props) => props.theme.card.background};
+
+        > div {
+          padding-right: 0;
+        }
+      }
+    }
+  }
+`
+
+const ArrowIcon = styled(ChevronDownIcon)`
+  margin-left: 6px;
+  transition: 0.5s;
+
+  &.toggle {
+    transform: rotate(180deg);
+  }
+`
+const CellInner = styled.div`
+  padding: 5px 0px;
+  display: flex;
+  width: 100%;
+  align-items: center;
+  padding-right: 16px;
+`
+
+const TableContainer = styled.div<{ showGradient: boolean }>`
+  position: relative;
+  &:after {
+    transition: 0.3s;
+    opacity: ${(props) => (props.showGradient ? '1' : '0')};
+    display: block;
+    content: "";
+    width: 14px;
+    height: 100%;
+    position: absolute;
+    top: 0;
+    right: 0;
+    background: linear-gradient(270deg, #E9EAEB 0%, rgba(233, 234, 235, 0) 100%);
+    z-index: 100;
+
+  }
+  ${({ theme }) => theme.mediaQueries.sm} {
+    &:after {
+      display: none;
+    }
+  }
+}
+`
+
+const columns = ColumnsDef.map((column) => ({
+  id: column.id,
+  name: column.name,
+  label: column.normal,
+  sort: (a: RowType<RowData>, b: RowType<RowData>) => {
+    switch (column.name) {
+      case 'farm':
+        return b.id - a.id
+      case 'apr':
+        if (a.original.apr.value && b.original.apr.value) {
+          return Number(a.original.apr.value) - Number(b.original.apr.value)
+        }
+
+        return 0
+      case 'details':
+        if (a.original.details.liquidity && b.original.details.liquidity) {
+          return a.original.details.liquidity - b.original.details.liquidity
+        }
+
+        return 0
+      case 'earned':
+        return a.original.earned.earnings - b.original.earned.earnings
+      case 'staked':
+        if (a.original.staked.userData && b.original.staked.userData) {
+          return new BigNumber(a.original.staked.userData.stakedBalance).comparedTo(
+            new BigNumber(b.original.staked.userData.stakedBalance),
+          )
+        }
+
+        return 0
+      default:
+        return 1
+    }
+  },
+  sortable: column.sortable,
+}))
+
+export default React.forwardRef((props: ITableProps) => {
+  const scrollBarEl = useRef<HTMLDivElement>(null)
+  const tableWrapperEl = useRef<HTMLDivElement>(null)
+  const [tableWidth, setTableWidth] = useState(600)
+  const [visibleScroll, setVisibleScroll] = useState(true)
+  const [showGradient, setShowGradient] = useState(true)
+  const TranslateString = useI18n()
+  const { data } = props
+
+  const renderSortArrow = (column: any): JSX.Element => {
+    if (column.sorted && column.sorted.on) {
+      if (column.sorted.asc) {
+        return <ArrowIcon className="toggle" color="secondary" />
+      }
+
+      return <ArrowIcon color="secondary" />
+    }
+
+    return null
+  }
+
+  const { headers, rows, toggleSort } = useTable(columns, data, { sortable: true, sortColumn: 'farm' })
+
+  useEffect(() => {
+    if (scrollBarEl && scrollBarEl.current) {
+      let isScrolling1 = false
```

Comment:
this is probably the worst naming you can use for a variable. Someone who doesn't know you code should be able to understand what a variable contains just by reading the name